src/mscorlib/src/System/String.Manipulation.cs

```diff
@@ -951,7 +951,173 @@ public String Remove(int startIndex, int count)
             Contract.EndContractBlock();
 
             return Substring(0, startIndex);
-        }   
+        }
+
+        public string Replace(string oldValue, string newValue, bool ignoreCase, CultureInfo culture)
+        {
+            Contract.Ensures(Contract.Result<String>() != null);
+            Contract.EndContractBlock();
+
+            return ReplaceCore(oldValue, newValue, culture, ignoreCase ? CompareOptions.IgnoreCase : CompareOptions.None);
+        }
+
+        public string Replace(string oldValue, string newValue, StringComparison comparison)
```

Comment:
The parameter should be named `comparisonType` to match the naming used everywhere else.